modules/machine-health-checks-about.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * master/deploying-machine-health-checks.adoc
+
+[id='machine-health-checks-about-{context}']
+= About `MachineHealthChecks`
+
+`MachineHealthChecks` are a specialized implementation of the `MachineAutoscaler`
+that automatically repairs unhealthy `Machines` in a particular `MachinePool`.
+
+To monitor machine health, you create a resource to define the
+configuration for a controller. You set a condition to check for, such as
+staying in the `NotReady` status for 15 minutes or displaying a permanent condition
+in the node-problem-detector, and a label for the set of machines to monitor.
+
+[NOTE]
+====
+You cannot apply a `MachineHealthCheck` to a machine with the master role.
+====
+
+The controller that observes a `MachineHealthCheck` resource checks for the status
+that you defined. If a machine fails the health check, it is automatically deleted
+and a new one is created to take its place. When a machine is deleted, you
+see <this kind of > event. The controller uses the same logic as the
+link:https://kubernetes.io/docs/concepts/architecture/nodes/#node-controller[Kubernetes node controller]
+to limit disruptive impact of the machine deletion.
```

Comment:
Not sure if this is completely valid assertion. @enxebre WDYT? Recently, we have merged node draining logic into aws actuator machine controller. So by default, when a machine is deleted and the machine is linked to a node, the node is drained (evicted) before it is deleted (e.g. in case pod disruption budget for pods is defined).

This might refer more to a fact that only a single machine is reconciled at a given time.

Reply:
agreed with @ingvagabund. This is not correct. Today the controller does not have a global vision of all machines hence it does not limit disruptive impact of the machine deletion

Reply:
I think I got that information from some data from November. Thank you for pointing out that it's out of date!

Is there anything specific that you think a user would need to know about machine deletion at this point? Just that the node's drained by evicting all the pods before it's deleted?